xlators/features/trash/src/trash.c

```diff
@@ -137,7 +137,15 @@ remove_trash_path(const char *path, gf_boolean_t internal, char **rem_path)
     if (internal)
         *rem_path = strchr(*rem_path + 1, '/');
 }
-
+int
+check_pathbuf(xlator_t *this, char *pathbuf)
+{
+    if (!pathbuf) {
+        gf_log(this->name, GF_LOG_DEBUG, "Inode path not found");
+        return EINVAL;
+    }
+    return 0;
+}
```

Comment:
When I was talking about moving common code to a single function I was referring to something like this (I've not tested it, just a quick and dirty copy&paste of code):

```suggestion
static int32_t
check_pathbuf(xlator_t *this, call_frame_t *frame, char **pathbuf, int32_t *match)
{
    trash_private_t *priv = this->private;
    trash_local_t *local;

    *pathbuf = NULL;
    inode_path(loc->inode, NULL, pathbuf);
    if (!pathbuf) {
        gf_log(this->name, GF_LOG_DEBUG, "Inode path not found");
        return -EINVAL;
    }
    *match = check_whether_eliminate_path(priv->eliminate, *pathbuf);
    if ((strncmp(pathbuf, priv->newtrash_dir, strlen(priv->newtrash_dir)) == 0) ||
        match) {
        GF_FREE(*pathbuf);
        return 0;
    }

    local = mem_get0(this->local_pool);
    if (!local) {
        GF_FREE(*pathbuf);
        return -ENOMEM;
    }

    frame->local = local;
    
    return 1;
}
```

And then remove everything done here from the code in the callers.

Reply:
Updated the code and also think that use of match can be reduced so tried to log in function itself by using char *name.Also there is redundancy as using loc->inode and loc->name in arguments but have to do in this way because in trash_ftruncate we are checking for fd.Local is passed by reference because there is use of it rather than declaring locally.
Suggest me if more optimization can be done.